sections/futures/TradeCrossMargin/TradeCrossMargin.tsx

```diff
@@ -0,0 +1,154 @@
+
+	const [leverageSide, setLeverageSide] = useRecoilState(leverageSideState);
+	const { crossMarginAddress, crossMarginAvailable } = useRecoilValue(futuresAccountState);
+	const selectedAccountType = useRecoilValue(futuresAccountTypeState);
+
+	const walletAddress = useRecoilValue(walletAddressState);
+	const { susdSize } = useRecoilValue(tradeSizeState);
+
+	const { onTradeAmountSUSDChange, maxUsdInputAmount } = useFuturesContext();
+
+	const [percent, setPercent] = useState(0);
+	const [usdAmount, setUsdAmount] = useState(susdSize);
+	const [showOnboard, setShowOnboard] = useState(false);
+	const [openTransferModal, setOpenTransferModal] = useState<'deposit' | 'withdraw' | null>(null);
+
+	// eslint-disable-next-line
+	const onChangeMarginPercent = useCallback(
+		(value, commit = false) => {
+			setPercent(value);
+			const fraction = value / 100;
+			const usdAmount = maxUsdInputAmount.mul(fraction).toString();
+			const usdValue = Number(usdAmount).toFixed(0);
+			setUsdAmount(usdValue);
+			onTradeAmountSUSDChange(usdValue, commit);
+		},
+		[onTradeAmountSUSDChange, maxUsdInputAmount]
+	);
+
+	useEffect(() => {
+		if (susdSize !== usdAmount) {
+			if (!susdSize || maxUsdInputAmount.eq(0)) {
+				setPercent(0);
+				return;
+			}
+
+			const percent = wei(susdSize).div(maxUsdInputAmount).mul(100).toNumber();
+			setPercent(Number(percent.toFixed(2)));
+		}
+		// eslint-disable-next-line
+	}, [susdSize]);
+
+	return (
+		<>
+			<CrossMarginOnboard onClose={() => setShowOnboard(false)} isOpen={showOnboard} />
+
+			{!walletAddress ? (
+				<MessageContainer>{t('futures.market.trade.cross-margin.connect-wallet')}</MessageContainer>
+			) : !crossMarginAvailable ? (
+				<CrossMarginUnsupported />
+			) : !crossMarginAddress ? (
+				<CreateAccount onShowOnboard={() => setShowOnboard(true)} />
+			) : (
+				<>
+					{!isMobile && <MarketsDropdown />}
+
+					<TradePanelHeader
+						accountType={selectedAccountType}
+						buttons={[
+							{
+								i18nTitle: 'futures.market.trade.button.deposit',
+								Icon: DepositArrow,
+								onClick: () => setOpenTransferModal('deposit'),
+							},
+							{
+								i18nTitle: 'futures.market.trade.button.withdraw',
+								Icon: WithdrawArrow,
+								onClick: () => setOpenTransferModal('withdraw'),
+							},
+						]}
+					/>
+					{}
+					<MarginInfoBox />
+					<OrderSizing />
+					<SliderRow>
+						<StyledSlider
+							minValue={0}
+							maxValue={100}
+							step={1}
+							defaultValue={percent}
+							value={percent}
+							onChange={(_, value) => onChangeMarginPercent(value, false)}
+							onChangeCommitted={(_, value) => onChangeMarginPercent(value, true)}
+							marks={[
+								{ value: 0, label: `0%` },
+								{ value: 100, label: `100%` },
+							]}
+							valueLabelDisplay="on"
+							valueLabelFormat={(v) => `${v}%`}
+							$currentMark={percent}
+						/>
+					</SliderRow>
+					<PositionButtons selected={leverageSide} onSelect={setLeverageSide} />
```

Comment:
Changing the position side doesn't trigger a new preview. This means that if you change the leverage side last, then click open position, the actual trade will not match the button:
<img width="945" alt="image" src="https://user-images.githubusercontent.com/10401554/188018665-5fdc699e-e56b-4610-a279-7c31181da1f5.png">

Reply:
fixed